packages/engine/lib/stateful/src/message/payload.rs

```diff
@@ -0,0 +1,30 @@
+use serde::{Deserialize, Deserializer, Serialize};
+
+fn value_or_string_array<'de, D>(deserializer: D) -> Result<Vec<String>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum ValueOrStringArray {
+        String(String),
+        Number(i64),
+        Vec(Vec<String>),
+    }
+
+    match ValueOrStringArray::deserialize(deserializer)? {
+        ValueOrStringArray::String(s) => Ok(vec![s]),
+        ValueOrStringArray::Number(i) => Ok(vec![i.to_string()]),
+        ValueOrStringArray::Vec(v) => Ok(v),
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct GenericPayload {
```

Comment:
Would be good to throw a doc-comment on this as a drive-by, I have no idea what this is used for without going and finding where it's used.

Reply:
Yup, added a string. It's used for arbitrary data like JSON data for message data